testsuite/MDAnalysisTests/visualization/test_streamlines.py

```diff
@@ -53,6 +53,132 @@ def membrane_xtc(tmpdir_factory, univ):
     return str(tmp_xtc)
 
 
+def test_produce_list_indices_point_in_polygon_this_frame():
+    # Define two squares:
+    # square1 covers the area [0,1]x[0,1]
+    # square2 covers the area [2,3]x[2,3]
+    square1 = [(0, 0), (1, 0), (1, 1), (0, 1)]
+    square2 = [(2, 2), (3, 2), (3, 3), (2, 3)]
+
+    # Create a list of vertex coordinates (for two squares)
+    vertex_list = [square1, square2]
+
+    # Define points:
+    # Point [0.5, 0.5] lies inside square1.
+    # Point [1.5, 1.5] lies outside both squares.
+    # Point [2.5, 2.5] lies inside square2.
+    # Point [3.5, 3.5] lies outside both squares.
+    points = np.array([[0.5, 0.5], [1.5, 1.5], [2.5, 2.5], [3.5, 3.5]])
+
+    # Call the function under test.
+    result = streamlines._produce_list_indices_point_in_polygon_this_frame(
+        vertex_list, points
+    )
+
+    # np.where returns a tuple; thus for each square we expect:
+    # For square1: point index 0 is inside → (array([0]),)
+    # For square2: point index 2 is inside → (array([2]),)
+    expected = [(np.array([0]),), (np.array([2]),)]
+
+    # Check that each result matches the expected indices.
+    for res_tuple, exp_tuple in zip(result, expected):
+        np.testing.assert_array_equal(res_tuple[0], exp_tuple[0])
+
+
+def test_produce_list_centroids_empty():
+    # Simulate an empty index set for one square:
+    list_indices = [(np.array([]),)]
+    # Dummy particle coordinate array (won't be used since indices is empty)
+    pts = np.array([[0, 0], [1, 1]])
+    result = streamlines._produce_list_centroids_this_frame(list_indices, pts)
+    assert result == [None]
+
+
+def test_produce_list_centroids_single_square():
+    # Create an array of particle coordinates
+    pts = np.array([[0, 0], [1, 1], [2, 2], [3, 3]])
+    # Choose indices that pick points [1] and [3]
+    indices_tuple = (np.array([1, 3]),)
+    list_indices = [indices_tuple]
+    result = streamlines._produce_list_centroids_this_frame(list_indices, pts)
+    expected = np.average(pts[[1, 3]], axis=0)
```

Comment:
```suggestion
    expected = np.array([2., 2.])
```
It's better to test actual values instead of re-running through the function logic.

Reply:
Added actual values directly!